LiveSplit/UpdateManager/Updater.cs

```diff
@@ -101,54 +101,51 @@ public IEnumerable<string> GetChangeLog()
 
             public void PerformUpdate()
             {
+                string ConvertChangeUrlPartToPath(string xmlChangePath) => xmlChangePath.Replace('/', Path.DirectorySeparatorChar);
+
                 IList<Update> updates = Updates.Where(x => x.Version > Version).ToList();
-                List<string> addedFiles = new List<string>();
-                List<string> changedFiles = new List<string>();
-                List<string> removedFiles = new List<string>();
+                var addedFiles = new Dictionary<string, string>();
+                var changedFiles = new Dictionary<string, string>();
+                var removedFiles = new Dictionary<string, string>();
 
-                foreach (FileChange change in updates.SelectMany(x => x.FileChanges))
+                foreach (var change in updates.SelectMany(x => x.FileChanges))
                 {
-                    bool inAddedFiles = addedFiles.Contains(change.Path);
-                    bool inChangedFiles = changedFiles.Contains(change.Path);
-                    bool inRemovedFiles = removedFiles.Contains(change.Path);
-
                     string path = change.Path;
+                    string localPath = change.LocalPath;
```

Comment:
I think you can do the defaulting here, rather than in two places when adding or removing files.
```suggestion
                    string localPath = change.LocalPath ?? change.Path;
```